vendingMachine.js

```diff
@@ -1,68 +1,101 @@
+      if (matrixStock[i][j] <= 1) {
+        matrixProducts[i][j] = `${supline.repeat(4)}`;
+      }
+      itemTable = itemTable + `${squareItemTable}`;
+      //When the end of the row is identified, add the product code and stock data.
+      if (j === matrixProducts.length - 1) {
+        let imparsubtraction = 1;
+        //Calculate the required spaces on the left and right to organize the codes.
+        spaceLeft = SPACES.repeat(Math.round((spaceTable - (codeProduct)) / 2));
+        spaceRight = SPACES.repeat(Math.round((spaceTable - (codeProduct)) / 2) - imparsubtraction);
+        //Calculate the spaces required on the left and right to organize the stock of the product in column 1.
+        spaceLeft1 = SPACES.repeat(Math.round((spaceTable - (("Stock:" + productStock[i][j - 2]).length)) / 2))
+        spaceRight1 = SPACES.repeat((productStock[i][j - 2].length % 2 != 1) ? spaceLeft1.length : spaceLeft1.length - imparsubtraction); // We use a ternary operator to find out whether the length of the product stock is even or odd.
+        //Calculate the spaces required on the left and right to organize the stock of the product in column 2.
+        spaceLeft2 = SPACES.repeat(Math.round((spaceTable - (("Stock:" + productStock[i][j - 1]).length)) / 2))
+        spaceRight2 = SPACES.repeat((productStock[i][j - 1].length % 2 != 1) ? spaceLeft2.length : spaceLeft2.length - imparsubtraction);
+        //Calculate the spaces required on the left and right to organize the stock of the product in column 2.
+        spaceLeft3 = SPACES.repeat(Math.round((spaceTable - (("Stock:" + productStock[i][j]).length)) / 2))
+        spaceRight3 = SPACES.repeat((productStock[i][j].length % 2 != 1) ? spaceLeft3.length : spaceLeft3.length - imparsubtraction);
+        //Organize the results in the coordinates and stock variables, to add it to the itemTable variable and print the final shade.
+        coord = `|${spaceLeft}${(i + 1)}.${(j - 1)}${spaceRight}|${spaceLeft}${(i + 1)}.${(j )}${spaceRight}|${spaceLeft}${(i + 1)}.${(j + 1)}${spaceRight}|`;
+        stock = `|${spaceLeft1}Stock:${matrixStock[i][j - 2]}${spaceRight1}|${spaceLeft2}Stock:${matrixStock[i][j - 1]}${spaceRight2}|${spaceLeft3}Stock:${matrixStock[i][j]}${spaceRight3}|`;
+        itemTable = itemTable + `|\n${stock}\n${coord}\n|${topLine}\n`;
       }
     }
   }
-  console.log(result);
+  console.log(itemTable);
 }
 
-function productStock(arr3, a, b) {
-  if (a <= arr3.length & b <= arr3.length) {
-    let newStock = arr3[a][b];
-    if (newStock !== 0) {
+function showStock(matrixStock, row, column) {
+  if (row <= matrixStock.length & column <= matrixStock.length) {
+    let newStock = matrixStock[row][column];
+    //When the product stock is different from 0, it shows the quantity by subtracting, otherwise it prints a message.
+    if (newStock != 0) {
       newStock = newStock - 1;
-      arr3[a][b] = newStock;
+      matrixStock[row][column] = String(newStock);

```

Comment:
why do you use String(...) method instead to use a simple "" string concat?

Reply:
Error fixed